### PR TITLE
Clean up feature flag protecting use of Bundler v2

### PR DESCRIPTION
**Pull request by brrygrdn**
Update dependabot-bundler to support V2 by default

### CODE REVIEW

File: bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb [hunk 1]
```diff
@@ -445,18 +445,6 @@
...
-          let(:dependency_files) do
-            bundler_project_dependency_files("imports_gemspec_version_clash_old_required_ruby_no_lockfile")
-          end
-          let(:current_version) { "3.0.1" }
-
-          it "ignores the minimum ruby version in the gemspec" do
-            expect(resolver.latest_resolvable_version_details[:version]).
-              to eq(Gem::Version.new("7.2.0"))
-          end
-        end
```
Comment (brrygrdn) [inline bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb:458]: This test no longer applies, absent a lockfile we will always use Bundler 2

File: bundler/spec/dependabot/bundler/update_checker_spec.rb [hunk 2]
```diff
@@ -1367,15 +1367,6 @@
...
         end
 
-        it "raises a helpful error", :bundler_v1_only do
-          expect { checker.latest_resolvable_version }.
-            to raise_error do |error|
-              expect(error).to be_a(Dependabot::GitDependenciesNotReachable)
-              expect(error.dependency_urls).
-                to eq(["git://github.com/fundingcircle/prius.git"])
-            end
-        end
```
Comment (brrygrdn) [inline bundler/spec/dependabot/bundler/update_checker_spec.rb:1377]: This test no longer applies, absent a lockfile we will always use Bundler 2

**Review by jurre — APPROVED** [review]

File: bundler/lib/dependabot/bundler/helpers.rb [hunk 3]
```diff
@@ -16,9 +16,7 @@ module Helpers
       BUNDLER_MAJOR_VERSION_REGEX = /BUNDLED WITH\s+(?<version>\d+)\./m.freeze
 
       # NOTE: options is a manditory argument to ensure we pass it from all calling classes
```
Comment (feelepxyz) [inline bundler/lib/dependabot/bundler/helpers.rb:18]:
```suggestion
```

**Comment by feelepxyz** [comment]
Should we not also jank out `bundler_2_available` from the test setup? https://github.com/dependabot/dependabot-core/blob/main/bundler/spec/dependabot/bundler/update_checker_spec.rb#L20-L22

**Comment by brrygrdn** [comment]
@feelepxyz Good point, I forgot about that. Will update.

**Review by feelepxyz — APPROVED** [review]
🎉

**Pull request merged**